src/main.cpp

```diff
@@ -361,37 +362,60 @@ int main(int argc, char** argv) {
 	int numFilesProcessed = 0;
 	int numErrors = 0;
 	if (fs::is_directory(input) == true) {
+		//Build files list
+		std::deque<fs::path> files_list;
 		std::cout << "We're going to be operating in a directory. dir:" << fs::absolute(input) << std::endl;
 		if (recursive_directory_iterator) {
 			for (auto & inputFile : fs::recursive_directory_iterator(input)) {
 				if (!fs::is_directory(inputFile)) {
-					numFilesProcessed++;
-					try {
-						convert_file(convInfo, inputFile, output);
-					}
-					catch (const std::exception& e) {
-						numErrors++;
-						std::cout << e.what() << std::endl;
-					}
+					files_list.push_back(inputFile);
 				}
 			}
 		}
 		else {
 			for (auto & inputFile : fs::directory_iterator(input)) {
 				if (!fs::is_directory(inputFile)) {
-					numFilesProcessed++;
-					try {
-						convert_file(convInfo, inputFile, output);
-					}
-					catch (const std::exception& e) {
-						numErrors++;
-						std::cout << e.what() << std::endl;
-					}
+					files_list.push_back(inputFile);
 				}
 			}
 		}
 
+		//Proceed by list
+		double timeAvg = 0.0;
+		int files_count = static_cast<int>(files_list.size());
+		for (auto &fn : files_list) {
+			++numFilesProcessed;
+			double time_file_start = getsec();
 
+			std::cout << "[" << numFilesProcessed << "/" << files_count << "] " << fn.filename();
```

Comment:
A linebreak `<< std::endl;` is missing here.

Reply:
It is possible and so, but initially in my version there were no new lines - everything was displayed in 1 line:
```
//std::cout << std::flush << "\r" << numFilesProcessed << "/" << files_count << ": " << input_file << " (elapsed: ";
```

Reply:
Well it produces this currently:
![image](https://user-images.githubusercontent.com/1492505/33574871-0c5f2706-d93b-11e7-8f31-36f28cc25dc2.png)
Doesn't look very nice

Reply:
yes, in that case it will not look very nice, you're right ...